odoo/addons/test_impex/tests/test_load.py

```diff
@@ -567,12 +567,33 @@ def test_by_name(self):
         name1 = dict(record1.name_get())[record1.id]
         name2 = dict(record2.name_get())[record2.id]
 
+        # preheat the oven
+        for _ in range(5):
+            self.env.cr.execute('SAVEPOINT xxx')
+            self.import_(['value'], [[name1], [name1], [name2]])
+            self.env.cr.execute('ROLLBACK TO SAVEPOINT xxx')
+            self.env.cr.execute('RELEASE SAVEPOINT xxx')
+
+        queries_before = self.cr.sql_log_count
         result = self.import_(['value'], [
             # import by name_get
             [name1],
             [name1],
             [name2],
         ])
+
+        # 1 x SAVEPOINT model_load
+        # 3 x name_search
+        # 1 x SAVEPOINT
+        # 3 x insert
+        # 1 x RELEASE SAVEPOINT
+        # => 9
+        expected_queries = 9
+        self.assertEqual(
+            self.cr.sql_log_count - queries_before, expected_queries,
+            "batched import w/name_get should only have %s queries" % expected_queries
+        )
```

Comment:
You can actually use `assertQueryCount`.

Reply:
BTW, how many queries do you save in this case?

Reply:
> You can actually use `assertQueryCount`.

Tru

> BTW, how many queries do you save in this case?

I'd have to re-check but I'd think 2 per record (so 4 in this case), as the unbatching means one savepoint/release per record being inserted, instead of just one for the entire thing.